File: fc-chat/demo/part3/ch3/05-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/entity/ChannelEntity.java
package net.prostars.messagesystem.entity;

import jakarta.persistence.*;
import java.util.Objects;
import java.util.UUID;

@Entity
@Table(name = "channel")
public class ChannelEntity extends BaseEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "channel_id")
  private Long channelId;

  @Column(name = "title", nullable = false)
  private String title;

  @Column(name = "invite_code", nullable = false)
  private String inviteCode;

  @Column(name = "head_count", nullable = false)
  private int headCount;

  public ChannelEntity() {}

  public ChannelEntity(String title, int headCount) {
    this.title = title;
    this.headCount = headCount;
    this.inviteCode = UUID.randomUUID().toString().replace("-", "");
  }

  public Long getChannelId() {
    return channelId;
  }

  public String getTitle() {
    return title;
  }

  public String getInviteCode() {
    return inviteCode;
  }

  public int getHeadCount() {
    return headCount;
  }

  public void setHeadCount(int headCount) {
    this.headCount = headCount;
  }

  @Override
  public boolean equals(Object o) {
    if (o == null || getClass() != o.getClass()) return false;
    ChannelEntity that = (ChannelEntity) o;
    return Objects.equals(channelId, that.channelId);
  }

  @Override
  public int hashCode() {
    return Objects.hashCode(channelId);
  }

  @Override
  public String toString() {
    return "ChannelEntity{channelId=%d, title='%s', inviteCode='%s', headCount=%d}"
        .formatted(channelId, title, inviteCode, headCount);
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/entity/ChannelEntity.kt
package com.yscorp.withpush.messagesystem.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "channel")
class ChannelEntity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "channel_id")
    val channelId: Long? = null,

    @Column(name = "title", nullable = false)
    var title: String,

    @Column(name = "invite_code", nullable = false)
    var inviteCode: String? = null,

    @Column(name = "head_count", nullable = false)
    var headCount: Int = 0,

    ) : BaseEntity() {

    constructor(title: String, headCount: Int) : this(
        title = title,
        inviteCode = UUID.randomUUID().toString().replace("-", ""),
        headCount = headCount
    ) {
    }

    override fun equals(o: Any?): Boolean {
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ChannelEntity
        return channelId == that.channelId
    }

    override fun hashCode(): Int {
        return Objects.hashCode(channelId)
    }

    override fun toString(): String {
        return "ChannelEntity{channelId=%d, title='%s', inviteCode='%s', headCount=%d}"
            .formatted(channelId, title, inviteCode, headCount)
    }
}
